h2o-algos/src/test/java/hex/glm/GLMTest.java

```diff
@@ -109,6 +109,132 @@ private void checkScore(long rid, double [] predictions, double [] outputs){
+  }
+  
+  public void testCoeffs(Family family, String fileName, String responseColumn) {
+    try {
+      Scope.enter();
+      Frame train = parse_test_file(fileName);
+      // set cat columns
+      int numCols = train.numCols();
+      int enumCols = (numCols-1)/2;
+      for (int cindex=0; cindex<enumCols; cindex++) {
+        train.replace(cindex, train.vec(cindex).toCategoricalVec()).remove();
+      }
+      int response_index = numCols-1;
+      if (family.equals(Family.binomial) || (family.equals(Family.multinomial))) {
+        train.replace((response_index), train.vec(response_index).toCategoricalVec()).remove();
+      }
+      DKV.put(train);
+      Scope.track(train);
+      
+      GLMParameters params = new GLMParameters(family);
+      params._standardize=true;
+      params._response_column = responseColumn;
+      params._train = train._key;
+      GLMModel glm = new GLM(params).trainModel().get();
+      Scope.track_generic(glm);
+      
+      // standardize numerical columns of train
+      int numStart = enumCols;  // start of numerical columns
+      int[] numCols2Transform = new int[enumCols];
+      double[] colMeans = new double[enumCols];
+      double[] oneOSigma = new double[enumCols];
+      int countIndex = 0;
+      HashMap<String, Double> cMeans = new HashMap<>();
+      HashMap<String, Double> cSigmas = new HashMap<>();
+      String[] cnames = train.names();
+      for (int cindex = numStart; cindex < response_index; cindex++) {
+        numCols2Transform[countIndex]=cindex;
+        colMeans[countIndex] = train.vec(cindex).mean();
+        oneOSigma[countIndex] = 1.0/train.vec(cindex).sigma();
+        cMeans.put(cnames[cindex], colMeans[countIndex]);
+        cSigmas.put(cnames[cindex], train.vec(cindex).sigma());
+        countIndex++;
+      }
+
+      params._standardize = false;  // build a model on non-standardized columns with no standardization.
+      GLMModel glmF = new GLM(params).trainModel().get();
+      Scope.track_generic(glmF);
+
+      HashMap<String, Double> coeffSF = glmF.coefficients(true);
+      HashMap<String, Double> coeffF = glmF.coefficients();
+      if (family.equals(Family.multinomial)) {
+        double[] interPClass = new double[glmF._output._nclasses];
+        for (String key : coeffSF.keySet()) {
+          double temp1 = coeffSF.get(key);
+          double temp2 = coeffF.get(key);
+          if (Math.abs(temp1 - temp2) > 1e-6) { // coefficient same for categoricals, different for numericals
+            String[] coNames = key.split("_");
+            if (!(coNames[1].equals("Intercept"))) {  // skip over intercepts
+              String colnames = coNames[1];
+              interPClass[Integer.valueOf(coNames[0])] += temp2 * cMeans.get(colnames);
+              temp2 = temp2 * cSigmas.get(colnames);
+            }
+            assert Math.abs(temp1 - temp2) < 1e-6 : "Expected coefficients for " + coNames[1] + " is " + temp1 + " but actual " + temp2;
```

Comment:
Can you exchange the lines 182 and l83.  It attempts to compare intercept before it is ready